tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -233,27 +222,120 @@ std::string toFrameId(const TangoCoordinateFrameType& tango_frame_type) {
   }
   return string_frame_type;
 }
+// Converts TangoCameraIntrinsics to sensor_msgs::CameraInfo.
+// @param camera_intrinsics, TangoCameraIntrinsics to convert.
+// @param camera_info, the output CameraInfo.
+void toCameraInfo(const TangoCameraIntrinsics& camera_intrinsics,
+                  sensor_msgs::CameraInfo* camera_info) {
+  camera_info->height = camera_intrinsics.height;
+  camera_info->width = camera_intrinsics.width;
+  camera_info->K = {camera_intrinsics.fx, 0., camera_intrinsics.cx,
+                    0., camera_intrinsics.fy, camera_intrinsics.cy,
+                    0., 0., 1.};
+  camera_info->R = {1., 0., 0., 0., 1., 0., 0., 0., 1.};
+  camera_info->P = {camera_intrinsics.fx, 0., camera_intrinsics.cx, 0.,
+                    0., camera_intrinsics.fy, camera_intrinsics.cy, 0.,
+                    0., 0., 1., 0.};
+  if (camera_intrinsics.camera_id == TangoCameraId::TANGO_CAMERA_FISHEYE) {
+    camera_info->distortion_model = sensor_msgs::distortion_models::PLUMB_BOB;
+    camera_info->D = {camera_intrinsics.distortion[0],
+        camera_intrinsics.distortion[1], camera_intrinsics.distortion[2],
+        camera_intrinsics.distortion[3], camera_intrinsics.distortion[4]};
+  } else if (camera_intrinsics.camera_id == TangoCameraId::TANGO_CAMERA_COLOR) {
+    camera_info->distortion_model = sensor_msgs::distortion_models::RATIONAL_POLYNOMIAL;
+    camera_info->D = {camera_intrinsics.distortion[0],
+        camera_intrinsics.distortion[1], 0., 0., camera_intrinsics.distortion[2]};
+  } else {
+    LOG(ERROR) << "Unknown camera ID: " << camera_intrinsics.camera_id;
+  }
+}
+// Compute fisheye distorted coordinates from undistorded coordinates.
+// The distortion model used by the Tango fisheye camera is called FOV and is
+// described in 'Straight lines have to be straight'.
+// See https://hal.inria.fr/inria-00267247/document.
+void ApplyFOVModel(
+    double xu, double yu, double w, double w_inverse, double two_tan_w_div_two,
+    double* xd, double* yd) {
+  double ru = sqrt(xu * xu + yu * yu);
+  const double epsilon = 1e-7;
+  if (w < epsilon) {
```

Comment:
The `else` statement seems ill conditioned for small `ru`. Perhaps change the `if` t
`w < epsilon || ru < epsilon`